### PR TITLE
Relation apply

### PR DESCRIPTION
**Pull request by lloricode**

### CODE REVIEW

**Comment by freekmurze** [comment]
Tests are failing for this PR.

Could you also add new tests that make sure that this new feature works? Also update the readme with some examples. I do feel like the implementation could be improved, seems kinda verbose now.

**Comment by lloricode** [comment]
@freekmurze  the est is just terminated, could you rerun it? 
then thanks, i will add some unit test for this, and wait for other collaborators feed back 😊

Thanks 😊

File: src/HasSlug.php [hunk 1]
```diff
@@ -122,6 +122,21 @@ protected function getSlugSourceString(): string
 
         $slugSourceString = collect($this->slugOptions->generateSlugFrom)
             ->map(function (string $fieldName) : string {
+
+                if (strpos($fieldName, '.') !== false) {
```
Comment (freekmurze) [inline src/HasSlug.php]: Could you extract this bit to it's own method. Bonus points for not having to use `continue`.

Comment (lloricode) [reply src/HasSlug.php]: ok :)

**Comment by lloricode** [comment]
travic-ci buld terminated 
:|

**Comment by freekmurze** [comment]
Restarted the build.

**Comment by freekmurze** [comment]
I've decided not to merge it in, because we don't need this functionality right now and the tests are failing.

**Comment by lloricode** [comment]
@lloricode its ok :)

**Comment by zefexdeveloper** [comment]
Any news on this? I really need this for a classified website I'm creating and I need to generate the slug based on relationship

**Comment by ilgala** [comment]
I saw that the `getSlugSourceString()` function also accepts `callable`. That's my workaround:

```php
/**
 * Get the options for generating the slug.
 */
public function getSlugOptions() : SlugOptions
{
    return SlugOptions::create()
        ->generateSlugsFrom(function (Model $model) {
            return $model->getKey()
              .'-'.$model->relation->field;
        })->saveSlugsTo('slug');
}
```

**Pull request closed without merging**